masu/external/downloader/aws/aws_report_downloader.py

```diff
@@ -74,10 +74,8 @@ def __init__(self, customer_name, auth_credential, cur_source, report_name=None,
         if not report:
             raise MasuConfigurationError('Cost and Usage Report definition not found.')
 
-        report_to_download = report.pop()
-        self.region = report_to_download['S3Region']
+        self.report = report.pop()
```

Comment:
I agree that the logic isn't very clear but I think it's covered with the MasuConfigurationError being thrown right above this.

It's gated by making sure that report_name is in the definitions returned.  report_name can be specified or a default value for the first one we come across.  If there are no reports in the bucket then self.report_name would be None.  Next it would not match any report_defs returned by get_cur_report_definitions.  So report would be None and we would throw the error.

Did I miss something here?